src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerJsonNetWithoutBinder.cs

```diff
@@ -0,0 +1,242 @@
+                                {
+                                    IInvocationOperation invocationOperation =
+                                        (IInvocationOperation)operationAnalysisContext.Operation;
+                                    if (jsonSerializerSymbol.Equals(invocationOperation.TargetMethod.ContainingType)
+                                        && SecurityHelpers.JsonSerializerDeserializationMethods.Contains(
+                                               invocationOperation.TargetMethod.Name))
+                                    {
+                                        lock (rootOperationsNeedingAnalysis)
+                                        {
+                                            rootOperationsNeedingAnalysis.Add(
+                                                (invocationOperation.GetRoot(), operationAnalysisContext.ContainingSymbol));
+                                        }
+                                    }
+                                },
+                                OperationKind.Invocation);
+
+                            operationBlockStartAnalysisContext.RegisterOperationAction(
+                                (OperationAnalysisContext operationAnalysisContext) =>
+                                {
+                                    IObjectCreationOperation objectCreationOperation =
+                                        (IObjectCreationOperation)operationAnalysisContext.Operation;
+                                    if (jsonSerializerSymbol.Equals(objectCreationOperation.Type))
+                                    {
+                                        lock (rootOperationsNeedingAnalysis)
+                                        {
+                                            rootOperationsNeedingAnalysis.Add(
+                                                (objectCreationOperation.GetRoot(), operationAnalysisContext.ContainingSymbol));
+                                        }
+                                    }
+                                },
+                                OperationKind.ObjectCreation);
+
+                            operationBlockStartAnalysisContext.RegisterOperationAction(
+                                (OperationAnalysisContext operationAnalysisContext) =>
+                                {
+                                    IReturnOperation returnOperation = (IReturnOperation)operationAnalysisContext.Operation;
+                                    if (jsonSerializerSymbol.Equals(returnOperation.Type))
+                                    {
+                                        rootOperationsNeedingAnalysis.Add(
+                                            (returnOperation.GetRoot(), operationAnalysisContext.ContainingSymbol));
+                                    }
+                                },
+                                OperationKind.Return);
```

Comment:
Maybe can remove this callback, cause it uses InterproceduralAnalysisKind.ContextSensitive?

Reply:
I don't think there's anything wrong with InterproceduralAnalysisKind.ContextSensitive, but I'm not using HazardousUsageEvaluatorKind.Return here, so this callback doesn't make sense.

---
In reply to: [307592781](https://github.com/dotnet/roslyn-analyzers/pull/2687#discussion_r307592781) [](ancestors = 307592781)